### PR TITLE
[RFC] Remove all references to JobActivity

### PR DESCRIPTION
**Pull request by mhinz**
The JobActivity event got replaced by callback functions provided to
jobstart() or termopen().

It got removed here:

https://github.com/neovim/neovim/commit/6e7757ad51dfe3b2de857ff8a8688718ff6115ac

---

This is a follow-up to: #2396

### CODE REVIEW

File: runtime/doc/eval.txt [hunk 1]
```diff
@@ -4061,9 +4061,8 @@ jobstop({job})						{Nvim} *jobstop()*
 		Stop a job created with |jobstart()| by sending a `SIGTERM`
 		to the corresponding process. If the process doesn't exit
 		cleanly soon, a `SIGKILL` will be sent. When the job is
-		finally closed, a |JobActivity| event will trigger with
-		`v:job_data[0]` set to `exited`.  See |job-control| for more
-		information.
+		finally closed, the exit handler provided to |jobstart()| or
+		|termopen()| will be run.  See |job-control| for more information.
```
Comment (Pyrohh) [inline runtime/doc/eval.txt]: Could you break the line after the period? It seems to be commonly done for sentences matching the scheme `See |help-doc| for more information`.

File: src/nvim/fileio.c [hunk 2]
```diff
@@ -6584,16 +6584,15 @@ apply_autocmds_group (
     fname = vim_strsave(fname);         /* make a copy, so we can change it */
   } else {
     sfname = vim_strsave(fname);
-    /* Don't try expanding FileType, Syntax, FuncUndefined, WindowID,
-     * ColorScheme, QuickFixCmd or JobActivity */
-    if (event == EVENT_FILETYPE
-        || event == EVENT_SYNTAX
+    // don't try expanding the following events
```
Comment (Pyrohh) [inline src/nvim/fileio.c]: :+1: much better

**Comment by Pyrohh** [comment]
Besides the small nits LGTM.

File: runtime/doc/eval.txt (hunk 1 above)
Comment (mhinz) [inline runtime/doc/eval.txt]: Ok

File: src/nvim/fileio.c (hunk 2 above)
Comment (mhinz) [inline src/nvim/fileio.c]: :)

**Comment by oni-link** [comment]
A comment about `v:job_data` in `test/functional/job/job_spec.lua` could also be removed.

**Comment by mhinz** [comment]
> A comment about v:job_data in test/functional/job/job_spec.lua could also be removed.

Done!

File: src/nvim/fileio.c [hunk 3]
```diff
@@ -6584,16 +6584,15 @@ apply_autocmds_group (
...
+    if (event == EVENT_COLORSCHEME
+        || event == EVENT_FILETYPE
         || event == EVENT_FUNCUNDEFINED
+        || event == EVENT_QUICKFIXCMDPOST
+        || event == EVENT_QUICKFIXCMDPRE
         || event == EVENT_REMOTEREPLY
         || event == EVENT_SPELLFILEMISSING
-        || event == EVENT_QUICKFIXCMDPRE
-        || event == EVENT_COLORSCHEME
-        || event == EVENT_QUICKFIXCMDPOST
```
Comment (splinterofchaos) [inline src/nvim/fileio.c]: Not sure why you reordered these (though the order is not a real concern).

**Comment by splinterofchaos** [comment]
LGTM :+1:

File: src/nvim/fileio.c (hunk 3 above)
Comment (mhinz) [inline src/nvim/fileio.c]: Just another of my small, sneaky "refactorings". :) I just sorted the entries for readability.

**Comment by Pyrohh** [comment]
I'll merge this (don't want 2 people trying to do the same thing at once).

**Comment by Pyrohh** [comment]
thanks @mhinz 

@justinmk I can't seem to get `ghrebase1` to work, how exactly do you use it? Here's what happens with #1446 for example:

```
~/src/neovim % ghrebase1 1446
Fetching origin
Enter passphrase for key 'bla bla bla':
Fetching upstream
remote: Counting objects: 9, done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 9 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (9/9), done.
From https://github.com/neovim/neovim
   8171fc4..9a363ca  master     -> upstream/master
Fetching oni-link
Fetching tarruda
Fetching bfredl
Fetching glts
Fetching mhinz
error: pathspec 'refs/pull/upstream/1446' did not match any file(s) known to git.
```

**Comment by fwalch** [comment]
You need to add a line to your `.git/config`:

```
[remote "upstream"]
	fetch = +refs/pull/*/head:refs/pull/upstream/*
```

**Comment by Pyrohh** [comment]
Thanks @fwalch.

**Pull request merged**